spec/defines/named_instance/moku_params_spec.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+require 'spec_helper'
+
+describe 'nebula::named_instance::moku_params' do
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) { os_facts }
+      let(:pre_condition) { 'include nebula::profile::moku' }
+
+      describe 'concat_fragments' do
+        let(:target) { "#{name} deploy init" }
+        let(:title) { "#{name} thishost" }
+
+        context 'for first-instance' do
+          let(:name) { 'first-instance' }
+          let(:params) do
+            {
+              instance: name,
+              url_root: '/first-instance-root',
+              users: %w[one_user another_user],
+              subservices: %w[one_subservice another_subservice],
+              source_url: 'git@github.com:mlibrary/first_invalid_default',
+              mysql_user: nil,
+              mysql_password: nil,
+              mysql_host: nil,
+              path: '/www-invalid/first-instance/app',
+              hostname: 'thishost',
+              datacenter: 'somedc',
+            }
+          end
+
+          it do
+            # Because of ADR-1, I'm setting this to pretty json instead
+            # of regular json. Since a human will be running moku init,
+            # it's easy to imagine that human wanting to look at the
+            # configuration file to check that it's set up right, and
+            # that's easier to do with readable json.
+            is_expected.to contain_concat_file(target).with(
+              path: '/etc/moku/init/first-instance.json',
+              format: 'json-pretty',
+            )
+          end
+
+          context 'with an init directory set' do
+            let(:init_directory) { "/#{Faker::Internet.domain_word}" }
+            let(:pre_condition) do
+              <<~EOT
+                class { 'nebula::profile::moku':
+                  init_directory => '#{init_directory}'
+                }
+              EOT
+            end
+
+            it do
+              is_expected.to contain_concat_file(target).with(
+                path: "#{init_directory}/first-instance.json",
+              )
+            end
+          end
+
+          {
```

Comment:
This is a little confusing because of the each at the end of such a large hash. What do you think of declaring this as a constant and then using it?